src/app/pages/ConnectorDetailsPage/ConfigurationTab/CommonStep.tsx

```diff
@@ -37,44 +37,59 @@ export const CommonStep: FC<CommonStepProp> = ({
 
   const [isSAUpdate, setIsSAUpdate] = React.useState(false);
   const [passwordHidden, setPasswordHidden] = React.useState<boolean>(true);
-
+  const [showEyeIcon, setShowEyeIcon] = React.useState<boolean>(false);
   const onNameChange = (val: any) => {
     onUpdateConfiguration('common', { ...configuration, name: val });
     val === '' ? changeIsValid(false) : changeIsValid(true);
   };
 
-  const onSAChange = (
-    val: string,
-    event: React.FormEvent<HTMLInputElement>
-  ) => {
-    if (event.currentTarget.id === 'connector-sa-id') {
-      onUpdateConfiguration('common', {
-        ...configuration,
-        service_account: {
-          client_id: val,
-          client_secret: configuration.service_account.client_secret,
-        },
+  const onSAChange = (clientId: string) => {
+    onUpdateConfiguration('common', {
+      ...configuration,
+      service_account: {
+        client_id: clientId,
+        client_secret: configuration.service_account.client_secret,
+      },
+    });
+    !isSAUpdate &&
+      setIsSAUpdate((prev) => {
+        if (!prev) {
+          changeIsValid(false);
+        }
+        return true;
       });
-      !isSAUpdate &&
-        setIsSAUpdate((prev) => {
-          if (!prev) {
-            changeIsValid(false);
-          }
-          return true;
-        });
+
+    if (
+      configuration.service_account.client_secret.includes('*') ||
```

Comment:
Is there a guarantee that a client secret definitely cannot contain a `*` character other than when we use a placeholder?

Reply:
@gashcrumb nice catch! I wasn't quite sure that secret can contain a `*` character.

Reply:
That's an option for sure, though some questions that raises for me are would that wind up getting added to the configuration object, and would that be a problem?  Thinking about it further, probably the placeholder value is really just a static token that this component cares about and is only a concern of the UI.   Maybe it could just be a global (to the component) const value.  If you do consider using some interim UI state object to hold these values, that placeholder value could live in there.

Reply:
_would that wind up getting added to the configuration object, and would that be a problem?_  that's what I am thinking probably just comparing 5 `*` `service_account.client_secret === '*****'` can solve. I don't think secret can just have 5 `*`  WDYT?